homeassistant/components/fritz/binary_sensor.py

```diff
@@ -82,13 +85,17 @@ def update(self) -> None:
         """Update data."""
         _LOGGER.debug("Updating FRITZ!Box binary sensors")
 
-        if self.entity_description.key == "is_connected":
-            self._attr_is_on = bool(self._fritzbox_tools.fritz_status.is_connected)
-        elif self.entity_description.key == "is_linked":
-            self._attr_is_on = bool(self._fritzbox_tools.fritz_status.is_linked)
-        elif self.entity_description.key == "firmware_update":
+        if self.entity_description.key == "firmware_update":
             self._attr_is_on = self._fritzbox_tools.update_available
             self._attr_extra_state_attributes = {
                 "installed_version": self._fritzbox_tools.current_firmware,
                 "latest_available_version": self._fritzbox_tools.latest_firmware,
             }
+
+        if self._fritzbox_tools.mesh_role == MeshRoles.SLAVE:
```

Comment:
For which entities and mesh role will this check be true?

Reply:
For all AVM devices (_fritz_tools) that have no mesh role (MeshRoles.NONE) or master role.

In a separate PR, I plan to rename both `self._fritz_tools` and `self._router` to a common name:` self._avm_device`

Simone

Reply:
And which entities does it apply to?

Reply:
Entities based on `is_connected` and `is_linked` are meaningful only for master/standalone routers.
All other devices, repeaters or slave routers, should not expose those even if the underlining API is available.

Simone

Reply:
Right, but we don't add those entities when mesh role is slave, so this case shouldn't happen unless mesh role is changed for the tool. And in that case it's better to remove the entities to make it consistent.

Reply:
Removed